src/providers/mdal/qgsmdalprovider.cpp

```diff
@@ -241,6 +241,22 @@ bool QgsMdalProvider::persistDatasetGroup( const QString &path,
   return false;
 }
 
+void QgsMdalProvider::reloadData()
+{
+  if ( mMeshH )
+    MDAL_CloseMesh( mMeshH );
+
+  QByteArray curi = dataSourceUri().toUtf8();
+  mMeshH = MDAL_LoadMesh( curi.constData() );
```

Comment:
maybe we can reuse setDataProvider() instead of this code?

Reply:
?

Reply:
just that this logic is already in the `setDataProvider()` method, so I am wondering if we cannot reuse it (maybe not)

Reply:
To avoid misunderstanding : do you refer to the code in QgsMeshLayer::reload() or to the code is commented ? (I have to ask you the same question for the previous comment).

For the QgsMdalProvider class, I think we can now factor the reloadData method and the constructor with a loadMesh method to avoid duplicate code.

Reply:
> maybe we can reuse setDataProvider() instead of this code?

setDataProvider() destruct the current provider and create a new one. It seems to be a quick way to reset and reload data.
But I think we should give the responsibility to the provider for choosing the way the data could be reloaded.
We don't know how futur providers will work to access data. It could be a problem if the provider instance is destructed and replaced by a new one, especially if the provider is link with a extern mechanism. What do you think about ?

Reply:
agree